test/test_common/utility.h

```diff
@@ -58,6 +58,17 @@ namespace Envoy {
     ADD_FAILURE() << "Unexpected exception: " << std::string(e.what());                            \
   }
 
+/*
+  Macro to use instead of EXPECT_DEATH when stderr is produced by a logger.
+  It temporarily installs stderr sink and restores the original logger sink after the test
+  completes and sdterrSink object goes of of scope.
```

Comment:
nit: Can you expand a bit on why this has to be used? I might also add this to STYLE.md since I think if we go this route it basically has to be used?